selector/selector.go

```diff
@@ -0,0 +1,42 @@
+package selector
+
+import (
+	"errors"
+
+	"github.com/micro/go-micro/v2/router"
+)
+
+var (
+	// DefaultSelector is the default selector
+	DefaultSelector = NewSelector()
+
+	// ErrNoneAvailable is returned by select when no routes were provided to select from
+	ErrNoneAvailable = errors.New("none available")
+)
+
+// Selector selects a route from a pool
+type Selector interface {
+	// Init a selector with options
+	Init(...Option) error
+	// Options the selector is using
+	Options() Options
+	// Select a route from the pool using the strategy
+	Select([]*router.Route) (*router.Route, error)
```

Comment:
It is simpler to define a "Route" or "Node" here so that router is not imported?

Reply:
I'd rather define an interface which is satisfied by the router.Route, however I think this is a wider issuer which we need to look at outside this PR.